app/api/resources/user.py

```diff
@@ -25,11 +25,22 @@ class UserList(Resource):
     @users_ns.doc('list_users')
     @users_ns.marshal_list_with(public_user_api_model)
     @users_ns.expect(auth_header_parser)
+    @users_ns.param(name="search", description="Search a user by the display name", _in="query")
     def get(cls):
         """
-        Returns list of all the users.
+        Returns two different results :
+        1. If `search` param is empty, it returns the list of all the users.
+        2. If `search` param is provided, it returns the user whose display name matches the search param.
         """
         user_id = get_jwt_identity()
+
+        search_param = request.args
+        # Verifying that search param is not empty
+        if search_param:
+            # Decoding the search param
+            search = search_param['search'].replace("%20"," ")
```

Comment:
Replace just space is not enough. We have special characters in French, German, Polish, Norwegian, Chinese, Arabic and many, many more languages. Please check if flask doesn't decode them automatically. If not use library like urllib.